spec/unit/puppet_webhook_spec.rb

```diff
@@ -1,9 +1,24 @@
 require 'spec_helper'
 
 describe PuppetWebhook do
+  it '/ returns 404 not found' do
+    get '/'
+    expect(last_response.status).to eq(404)
+    expect(last_response.body).to eq("You shall not pass! (page not found)\n")
+  end
+
   it '/heartbeat returns 200 ok and success message' do
     get '/heartbeat'
     expect(last_response).to be_ok
     expect(last_response.body).to eq('{"status":"success","message":"running"}')
   end
+
+  context 'github module endpoint' do
+    it '/module returns ok' do
+      stub_const('LOGGER', Logger.new('spec/fixtures/test.log'))
+      basic_authorize('puppet', 'puppet')
+      post '/module', { repository: { name: 'puppet-module'} }.to_json, { 'Content-Type' => 'application/json', 'X-Github-Event' => 'push' }
```

Comment:
You can also set headers this way.
```ruby
header 'Content-Type', 'application/json'
header 'X-GitHub-Event', 'push'
post '/module', { repository: { name: 'puppet-module'} }.to_json
```

Reply:
And this could use some assertion about the response.

Reply:
The post body is missing `owner: { login: 'login' }` so the parser chokes and the route never gets called.

Reply:
Should we just use our github fixture?
```
post '/module', read_fixture('github/update.json')
```